main.nf

```diff
@@ -1,234 +1,292 @@
+        --script_dir                        Location of project scripts (default: "\$baseDir/bin")
+
+        --help                              Print this help message.
+
+    Notes:
+        * Always quote paths that are parameters to nextflow e.g. '/path/to/file'
+
+    """
+}
+
+if (params.help || !params.run_folder){
+    helpMessage()
+    exit 0
+}
+
+workflow {
+
+    main:
+    get_run_folder(params.run_folder) | check_run_quality
+
+    publish:
+    check_run_quality.out.projectqc            to: "${params.result_dir}/multiqc_by_project"
+    check_run_quality.out.flowcellqc           to: "${params.result_dir}/multiqc_by_flowcell"
+
+}
+
+workflow.onComplete {
+	log.info ( workflow.success ? "\nDone! Open the reports in your browser.\n" : "Oops .. something went wrong." )
+}
+
+def get_run_folder(run_folder) {
+
+    Channel.value(file(run_folder))
+        .ifEmpty { "Error: No run folder (--run_folder) given."; exit 1 }
+
+}
+
+def get_project_and_reads(run_folder) {
+
+    Channel
+        .fromPath("${run_folder}/${params.bcl2fastq_outdir}/**.fastq.gz", maxDepth: 10 )
+        .filter( ~/.*_[^I]\d_001\.fastq\.gz$/ )
+        .ifEmpty { "Error: No fastq files found under ${run_folder}/ !\n"; exit 1 }
+        .map {
+            it.toString().indexOf('Undetermined') > 0 ?
+                ['NoProject', it] :
+                [(it.toString() =~ /^.*\/Unaligned\/([^\/]+)\/.*\.fastq\.gz$/)[0][1],it]
+        }
+
+}
+
+def combine_results_by_project (fastqc_results,fastq_screen_results) {
+
+    fastqc_results.mix(fastq_screen_results).groupTuple().map { it -> tuple(it[0],it[1][0].flatten(),it[1][1].flatten()) }
 
-params.additional_output_dir = ""
-if (params.additional_output_dir.length()) {
-    additional_output_dir = file(params.additional_output_dir)
 }
 
+workflow check_run_quality {
+
+    /* Workflow Graph
+
+    Runfolder -> InterOp                                -> MultiQCPerFlowcell
+        |     -> GetQCThresholds                        -> MultiQCPerFlowcell
+        |     -> GetMetaData                            -> MultiQCPerFlowcell + MultiQCPerProject
+        |
+        \ -> "Undeterminded/<Project>" -> FastQC        -> MultiQCPerFlowcell + MultiQCPerProject
+                                       -> FastqScreen   -> MultiQCPerFlowcell + MultiQCPerProject
+    */
+
+    get:
+        run_folder
+
+    main:
+        interop_summary(run_folder)
+        get_QC_thresholds(run_folder)
+        get_metadata(run_folder)
+        project_and_reads = get_project_and_reads(params.run_folder)
+        fastqc(project_and_reads)
+        fastq_screen(project_and_reads)
+        multiqc_per_flowcell( params.run_folder,
+            fastqc.out.map{ it[1] }.collect().ifEmpty([]),
+            fastq_screen.out.map{ it[1] }.collect().ifEmpty([]),
+            interop_summary.out.map{ it[1] }.collect().ifEmpty([]),
+            get_QC_thresholds.out.collect(),
```

Comment:
I think the case where get_QC_thresholds is empty must be handled here.

Reply:
Corrected (eventually). Collect was supposed to return a List object and ifEmpty failed with a runtime error, but it seems Collect now returns a dataflow object and ifEmpty works. Only the get_qc_thresholds should ever be empty. The other channels should throw an error before getting to multiqc if nothing is found.